### DISCUSSION TITLE
running userscripts in qutebrowser virtualenv

### DISCUSSION THREAD

**Discussion by cspence001** [qutebrowser/qutebrowser#7366] [Q&A]
I'm running qutebrowser in a virtualenv and launching with .venv/bin/python3 -m qutebrowser from the main application directory.
[venv install](https://github.com/qutebrowser/qutebrowser/blob/master/doc/install.asciidoc#installing-qutebrowser-with-virtualenv)
I want to run userscripts but am having trouble referencing them from the application browser after launching in vm. 
Initially I created a /userscripts directory as well as a config.py after the .qutebrowser/ was spawned in the initial launch (.venv/bin/python3 -m qutebrowser). I used this [template](https://github.com/SidharthArya/.qutebrowser/blob/master/config.py) as well as the localhost userscript [here](https://github.com/SidharthArya/.qutebrowser/blob/master/userscripts/localhost) as well as tried the initial solution [here](https://blog.sidhartharya.com/a-quicker-way-to-reach-localhost-with-qutebrowser/#solution). 

I'm running into an error:
ERROR: Userscript '/users/mYuSeRnAmE/.qutebrowser/userscripts/localhost' failed to start: execve: Exec format error
I'm guessing it has something to do with the shebang reference at the top of each of the userscripts. 

I've tried:
#!/bin/python3
#!/venv/bin/python3

and a few others that made less sense. Because I'm calling it from inside the venv already I'm assuming that the venv is left out and the exec variable is called with /bin/python3 to call the config.bind that references the localhost userscript for example (I've noticed the other userscripts in the example link each use different shebangs). My default shell is -zsh but I'm assuming that shouldn't matter because I'm calling from the virtualenv application and executing with python3 within. 

Is there another way to do this when running userscripts from the venv browser?

**Comment by toofar** [comment]
> execve: Exec format error
> I'm guessing it has something to do with the shebang reference 

Possibly, but just pointing to the wrong, or a non existent file, wont give that error. I just tried that and got `/bin/asdf: bad interpreter: No such file or directory`.

Are you editing the userscript on windows? I think this can happen if you have BOM at the start of the file. See https://stackoverflow.com/questions/42494853/standard-init-linux-go178-exec-user-process-caused-exec-format-error and https://unix.stackexchange.com/questions/381230/how-can-i-remove-the-bom-from-a-utf-8-file/412845#412845

**Reply by cspence001** [reply]
I actually just got it running a few minutes ago. It was #!/bin/bash which I thought I had tried. I used the very basic scripts [here](https://blog.sidhartharya.com/a-quicker-way-to-reach-localhost-with-qutebrowser/#solution) (running in venv and calling the config and userscripts/localhost in the .qutebrowser in my home directory). 

The error I get now is: 
ERROR: Load error: ERR_CONNECTION_REFUSED 
when trying to connect to localhost. It looks like by default it connects via chromium. I turned off my firewall (no restart). I have a few rules in little-snitch where I've blocked google connections but not chromium specifically so more than likely an app configuration.

**Comment by The-Compiler** [comment]
Moving this to a discussion as this is not a bug by any means.
